File: front-end/src/app/model/GSMController.ts
export class GSMController {
  id: number;
  roomID: number;
  phoneNumber: string;
  status: string;
  type: string;
  name: string;


  constructor(id: number, roomID: number, phoneNumber: string, status: string, type: string, name: string) {
    this.id = id;
    this.roomID = roomID;
    this.phoneNumber = phoneNumber;
    this.status = status;
    this.type = type;
    this.name = name;
  }
}

File: remohouse-core/src/main/java/org/circuitdoctor/core/model/GSMController.java
package org.circuitdoctor.core.model;

import lombok.*;
import javax.persistence.*;
import java.io.Serializable;
import javax.print.DocFlavor;
import javax.validation.constraints.*;

import static org.circuitdoctor.core.model.GSMStatus.OFF;

@EqualsAndHashCode(callSuper = true)
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
@Getter
@Setter
@Builder
@ToString
@Table(name="gsm_controller")
public class GSMController extends BaseEntity<Long> implements Serializable {
    @Column(nullable = false,unique = true)
    @NonNull
    @NotBlank(message = "phoneNumber is mandatory")
    @Digits(fraction=0,integer=10)
    private String phoneNumber;
    @Column(nullable = false)
    @Builder.Default
    private GSMStatus status=OFF;
    @Column(nullable = false)
    private String type;
    @ManyToOne(fetch = FetchType.EAGER)
    private Room room;
}
